flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreWrite.java

```diff
@@ -40,10 +44,12 @@ RecordWriter<T> createWriter(
     RecordWriter<T> createEmptyWriter(
             BinaryRowData partition, int bucket, ExecutorService compactExecutor);
 
-    /** Create a compact {@link RecordWriter} from partition, bucket and restore files. */
-    RecordWriter<T> createCompactWriter(
-            BinaryRowData partition,
-            int bucket,
-            ExecutorService compactExecutor,
-            List<DataFileMeta> restoredFiles);
+    /**
+     * Create a {@link Callable} compactor from partition, bucket.
+     *
+     * @param compactFiles input files of compaction. When it is null, will automatically read all
+     *     files of the current bucket.
+     */
+    Callable<CompactResult> createCompactWriter(
+            BinaryRowData partition, int bucket, @Nullable List<DataFileMeta> compactFiles);
```

Comment:
Just curious about which condition `compactFiles` is passed as `null`

Reply:
Actually, here we can just pass a null, but for the performance, we reuse compact files.